### PR TITLE
[swap_device] change lseek return type from int to off_t @open sesame 12/20 15:37

### PR DESCRIPTION
**Pull request by lhs8928**
- Return value of lseek is offset of requested position. The 4byte int type cannot represent more than 3GB so any request more than 3GB offset cause overflow.

Signed-off-by: hyeonseok lee <hs89.lee@samsung.com>

### CODE REVIEW

**Review by jijoongmoon — APPROVED** [review]
LGTM

File: nntrainer/tensor/swap_device.cpp [hunk 1]
```diff
@@ -108,11 +109,13 @@ void SwapDevice::putBuffer(void *ptr) {
 
   auto [offset, size] = allocated[ptr];
 
-  ret = lseek(fd, offset, SEEK_SET);
-  NNTR_THROW_IF(ret < 0, std::runtime_error) << "seek file: " << dev_path;
+  off_t off;
```
Comment (jihochu) [inline nntrainer/tensor/swap_device.cpp:112]: it's better to move line 107

File: nntrainer/tensor/swap_device.cpp [hunk 2]
```diff
@@ -108,11 +109,13 @@ void SwapDevice::putBuffer(void *ptr) {
...
+  off = lseek(fd, offset, SEEK_SET);
+  NNTR_THROW_IF(off < 0, std::runtime_error) << "seek file: " << dev_path;
 
-  ret = write(fd, ptr, size);
-  NNTR_THROW_IF(ret != size, std::runtime_error) << "write file: " << dev_path;
+  ssize_t len;
```
Comment (jihochu) [inline nntrainer/tensor/swap_device.cpp:116]: ditto.

**Review by DonghakPark — APPROVED** [review]
LGTM!

**Review by jihochu — APPROVED** [review]
LGTM.

File: nntrainer/tensor/swap_device.cpp (hunk 1 above)
Comment (lhs8928) [reply nntrainer/tensor/swap_device.cpp]: Done

File: nntrainer/tensor/swap_device.cpp (hunk 2 above)
Comment (lhs8928) [reply nntrainer/tensor/swap_device.cpp]: Done

**Pull request merged**